pytorch_lightning/trainer/trainer.py

```diff
@@ -928,7 +928,7 @@ def test(
         self.testing = True
 
         # If you supply a datamodule you can't supply test_dataloaders
-        if test_dataloaders and datamodule:
+        if test_dataloaders is not None and datamodule:
```

Comment:
should we apply the same checks to the other methods? e.g. in L869 for `validate` ?

Reply:
yes, I need to find all occurences